examples_tests/43A.SumAndCDFFilters/main.cpp

```diff
@@ -0,0 +1,99 @@
+#include <irrlicht.h>
+
+#include "../../../include/irr/asset/filters/CSummedAreaTableImageFilter.h"
+#include "../ext/ScreenShot/ScreenShot.h"
+
+using namespace irr;
+using namespace core;
+using namespace asset;
+using namespace video;
+
+
+int main()
+{
+	irr::SIrrlichtCreationParameters params;
+	params.Bits = 32;
+	params.ZBufferBits = 24;
+	params.DriverType = video::EDT_OPENGL;
+	params.WindowSize = dimension2d<uint32_t>(1600, 900);
+	params.Fullscreen = false;
+	params.Doublebuffer = true;
+	params.Vsync = true;
+	params.Stencilbuffer = false;
+
+	auto device = createDeviceEx(params);
+	if (!device)
+		return false;
+
+	device->getCursorControl()->setVisible(false);
+	auto driver = device->getVideoDriver();
+	auto assetManager = device->getAssetManager();
+	auto sceneManager = device->getSceneManager();
+
+	auto getSummedImage = [](core::smart_refctd_ptr<ICPUImage> image) -> core::smart_refctd_ptr<ICPUImage>
+	{
+		using SUM_FILTER = CSummedAreaTableImageFilter<true>;
+
+		core::smart_refctd_ptr<ICPUImage> newSumImage;
+		{
+			auto referenceImageParams = image->getCreationParameters();
+			auto referenceBuffer = image->getBuffer();
+			auto referenceRegions = image->getRegions();
+			auto referenceRegion = referenceRegions.begin();
+
+			auto newImageParams = referenceImageParams;
+			auto newCpuBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(referenceBuffer->getSize());
```

Comment:
you need to change format to something like RGBA_32_SFLOAT because the input are values in the [0,1] range, so a sum of all these values will be in [0,imageWidth*imageHeight] and you will also need the precision in the fractions, which makes UINT or INT formats unsuitable

Reply:
naturally the cpu-buffer also can be bigger

Reply:
so should I remove template parameter and make it use always `double`?

Reply:
solved